app/Services/Email.php

```diff
@@ -46,12 +47,15 @@ class Email
     /**
      * Constructor
      */
-    public function __construct(Message $message, Contest $contest, Competition $competition, $users)
+    public function __construct($resources)
     {
-        $this->message = $message;
-        $this->contest = $contest;
-        $this->competition = $competition;
-        $this->users = $users;
+        $this->message = $resources['message'];
+        $this->contest = $resources['contest'];
+        $this->competition = $resources['competition'];
+        $this->users = $resources['users'];
+        // @TODO - is there a better way of instantiating and pulling in the manager class?
+        // Maybe this email class needs to be a service instead.
+        // $this->manager = $manager;
```

Comment:
@weerd m not yet, sorry i was just putting this up there for eyes, but what i was trying to do was 

```
function __contruct($resources, Manager $manager) { }
```

but then it is not injecting the class magically, it expects the second param to be a Manager instance when I try to instantiate the `Email` class